### PR TITLE
WIP: Fix Julia 0.7 deprecation warnings 

### PR DESCRIPTION
**Pull request by lbenet**

### CODE REVIEW

File: src/intervals/set_operations.jl [hunk 1]
```diff
@@ -84,7 +84,7 @@ union(a::Interval, b::Interval) = hull(a, b)
 # union(a::Interval, b::Interval) = union(promote(a, b)...)
 
 
-"""
+@doc doc"""
```
Comment (dpsanders) [inline src/intervals/set_operations.jl]: Why does this need an `@doc`?

File: src/multidim/setdiff.jl [hunk 2]
```diff
@@ -33,7 +33,8 @@ end
 #
 #     end
 # end
-"""
+
+@doc doc"""
```
Comment (dpsanders) [inline src/multidim/setdiff.jl]: Also probably no longer needs `@doc`.

Comment (dpsanders) [reply src/multidim/setdiff.jl]: (After the change of `\` to `\setminus` in my PR from yesterday.)

**Comment by dpsanders** [comment]
LGTM -- thanks a lot for the heroic effort!

File: src/multidim/setdiff.jl (hunk 2 above)
Comment (lbenet) [reply src/multidim/setdiff.jl]: I guess the problem was indeed solved yesterday by merging #157. Let me rebase to current master and repush, to see how travis reacts.

**Comment by lbenet** [comment]
Tests are not passing. I'll try to fix the problems. 

An important part of the deprecation warnings now is:
```
┌ Warning: `setrounding` for `Float32` and `Float64` has been deprecated, and will not be available in future versions.
│   caller = setrounding(::getfield(IntervalArithmetic, Symbol("##37#38")){Float64}, ::Type{Float64}, ::RoundingMode{:Down}) at rounding.jl:156
└ @ Base.Rounding rounding.jl:156
```

Any suggestions?

**Comment by lbenet** [comment]
I just pushed another commit fixing some further deprecations for Julia 0.7. Yet, locally some tests are not passing, and I do not quite understand the problem. One concrete example is the following:
```julia
julia> using IntervalArithmetic

julia> @format full
6

julia> a = 0..3
Interval(0.0, 3.0)

julia> a^-2   # WRONG!! Should be --> Interval(0.1111111111111111, Inf)
Interval(0.11111111111111109, Inf)

julia> @which a^-2
^(a::Interval{Float64}, x::Integer) in IntervalArithmetic at /Users/benet/.julia/dev/IntervalArithmetic/src/intervals/functions.jl:10
```

Using the function that actually executes `a^-2` (IntervalArithmetic/src/intervals/functions.jl:10) I get the following:
```julia
julia> IntervalArithmetic.atomic(Interval{Float64}, IntervalArithmetic.big53(a)^-2)  # correct answer!!
Interval(0.1111111111111111, Inf)
```

Any idea why are we getting different results?
Any idea what is happening?

**Comment by dpsanders** [comment]
I had already noticed that problem and tried to find the issue but haven't managed to pin it down yet...

**Comment by lbenet** [comment]
There is (at least) one test that locally is not passing. I'm commenting this to first fix another problem, which I already encountered...

**Comment by lbenet** [comment]
The test I am trying to fix now (in Julia 0.6), which seems to be ok in master is the one making travis fail: `dist(a, b) <= eps(a)`.

**Comment by lbenet** [comment]
@dpsanders Locally, I get a broken test for current master in Julia 0.6 in `test/multidim_tests/multidim.jl`; can you confirm it? It is a test of the form `mid(X, 0.5)` for `X::IntervalBox`.

**Comment by lbenet** [comment]
(Broadcasting is driving me crazy in 0.7)

**Comment by lbenet** [comment]
Rebased to current master.

**Comment by dpsanders** [comment]
Is this ready to merge?

**Comment by lbenet** [comment]
Not quite yet; somehow the power function `^` is behaving oddly; at least that's one reason why tests are not passing in Julia 0.7...

**Comment by dpsanders** [comment]
I don't think that's anything to do with this PR though, so probably we should merge this and open an issue about ^.

**Comment by lbenet** [comment]
Ok, so go ahead and merge. In any case, I'll try to find out what the problem is.

**Comment by lbenet** [comment]
Rebased to current master.

**Comment by dpsanders** [comment]
Thanks a lot, we're getting closer!
Have you tried using FemtoCleaner?

**Comment by lbenet** [comment]
No, I haven't tried it yet.

**Pull request merged**